presto-spark-base/src/main/java/com/facebook/presto/spark/execution/HttpNativeExecutionTaskResultFetcher.java

```diff
@@ -208,12 +197,18 @@ public void run()
                     future.completeExceptionally(e);
                 }
             }
-            catch (ExecutionException | PrestoException e) {
-                client.abortResults();
-                future.completeExceptionally(e);
-            }
-            catch (TimeoutException e) {
-                if (++timeoutRetries >= MAX_HTTP_TIMEOUT_RETRIES) {
+            catch (ExecutionException e) {
+                try {
+                    throw e.getCause();
```

Comment:
why not just use `if (e.getCause() instanceof PageTransportErrorException)`? try-catch is more expensive than a if-else check.